bin/src/command/mod.rs

```diff
@@ -627,11 +627,12 @@ pub fn start(
         })
         .detach();
 
-        let saved_state = config.saved_state_path();
+        let saved_state_path = config.saved_state.clone();
+
         let mut server = CommandServer::new(fd, config, tx, command_rx, workers, accept_cancel_tx)?;
         server.load_static_application_configuration().await;
 
-        if let Some(path) = saved_state {
+        if let Some(path) = saved_state_path {
```

Comment:
I wonder about changing this if statement to something like this
```rust
if let Some(path) = config.saved_state.clone() {
```
as we do not need to store the value

Reply:
Sure !

Reply:
Wait, `config` is actually borrowed here on line 631: 

```rust
        let mut server = CommandServer::new(fd, config, tx, command_rx, workers, accept_cancel_tx)?;
```
I have to clone `saved_state` beforehand.